dsdump/XRMachOLibrary+SymbolDumper.mm

```diff
@@ -136,6 +152,51 @@ void print_symbol(XRMachOLibrary *object, struct nlist_64 * _Nonnull sym, uintpt
         return;
     }
     output_len += printf("0x%011llx ", override_addr ? *override_addr : sym->n_value);
+    
+    // Default color for symbol names
+    DSCOLOR symbolColor = DSCOLOR_CYAN;
+    
+    // Demangle and colorize symbols by languages/types
+    
+    // I. Demangle mode only (~nm -C option)
+    if (xref_options.demangle_mode && !xref_options.color) {
+        // Swfit symbols - Swift stable global mangling: '$s', Swift 4.2: '$S', Swift 4.0: '_T0'
+        // (Ref: https://github.com/apple/swift/blob/master/docs/ABI/Mangling.rst)
+        if (strnstr(chr, "_$s", 3) || strnstr(chr, "_swift", 6)) {
+            std::string outDemangledstring;
+            dshelpers::compact_demangle(chr, outDemangledstring);
+            chr = &outDemangledstring[0];
+        // C++ symbols
+        } else if (strnstr(chr, "__Z", 3)) {
+            chr = demangleCPP(chr);
+        }
+    // II. Catagolrize symbols and types with colors
+    } else if (xref_options.color) {
+        // Swfit symbols in Red
+        if (strnstr(chr, "_$s", 3) || strnstr(chr, "_swift", 6)) {
```

Comment:
More curious since I'll eventually be making a suite of unit tests: where have you seen the _swift symbols in the wild? How could I replicate them through source

Reply:
"_swift" are Swift Runtime C++ symbols (ref: [Runtime.md](https://github.com/apple/swift/blob/master/docs/Runtime.md))
You can found them in almost all swift binaries (e.g. macOS' Podcasts).

Maybe remove it for now to avoid confusions (C++ symbols for Swift Runtime library, not Swift symbols per se)